### DISCUSSION TITLE
How To Get Access Token via npsso

### DISCUSSION THREAD

**Discussion by dougcandido92** [achievements-app/psn-api#166] [General]
How to obtain the access token via npsso, because in the source code the access token is set fixed
![2024-06-03 18 58 48](https://github.com/achievements-app/psn-api/assets/19224177/8a9e48bd-74ba-4c2d-9b1b-75c19cc4c1be)

**Comment by wescopeland** [comment]
None of the headers here are hardcoded: https://github.com/achievements-app/psn-api/blob/main/src/authenticate/exchangeNpssoForCode.ts

Call `exchangeNpssoForCode()`.

https://psn-api.achievements.app/authentication/authenticating-manually#exchange-your-npsso-token
